File: php/src/Behavioral/Observer/DvdSubscriber.php
<?

#//DvdSubscriber - the observer

class DvdSubscriber
{

	public $name;

	function __construct($name)
	{
		$this->name = $name ;
	} 
   
	function newDvdRelease($release,$subscriberName)
	{
		return "Hello " . 
			$this->name . ", subscriber to the " . 
			$subscriberName . " DVD release list.  The new Dvd " . 
			$release->name . " will be released on " . 
			$release->releaseMonth . "/" . 
			$release->releaseDay . "/" . 
			$release->releaseYear . ".";
	}

	function updateDvdRelease($release,$subscriptionListName)
	{
		return "Hello " . 
			$this->name .  ", subscriber to the " .  
			$subscriptionListName .  " DVD release list.  The following DVDs release has been revised: " .  
			$release->name . " will be released on " .  
			$release->releaseMonth . "/" .  
			$release->releaseDay . "/" .  
			$release->releaseYear . ".";
	}   
}

File: go/behavioral/observer/DvdSubscriber.go
package observer

import (
	"fmt"
)

type DvdSubscriber struct {  
   Name string
}

func NewDvdSubscriber(name string) *DvdSubscriber {
	obj := new(DvdSubscriber)
	obj.Name = name 
	return obj
}

func (this *DvdSubscriber) NewDvdRelease(rel *DvdRelease , subname string) string {
	return fmt.Sprintf("Hello %s, subscriber to the %s DVD release list.\n" +
		"The new Dvd %s will be released on %d/%d/%d.",
			this.Name, subname, rel.DvdName, rel.Month, rel.Day, rel.Year )

}
   
func (this *DvdSubscriber ) UpdateDvdRelease (rel *DvdRelease , subscriptionListName string) string {
	return fmt.Sprintf("Hello %s, subscriber to the %s DVD release list.\n"  +
		"The following DVDs release has been revised: %s will be released on %d/%d/%d." ,
			this.Name, subscriptionListName, rel.DvdName, rel.Month , rel.Day , rel.Year)
}
